stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -584,24 +590,23 @@ async def execute_search(
             raise NotFoundError(f"Collections '{collection_ids}' do not exist")
 
         hits = es_response["hits"]["hits"]
-        items = (hit["_source"] for hit in hits)
+        items = (hit["_source"] for hit in hits[:limit])
 
         next_token = None
-        if hits and (sort_array := hits[-1].get("sort")):
-            next_token = urlsafe_b64encode(
-                ",".join([str(x) for x in sort_array]).encode()
-            ).decode()
-
-        # (1) count should not block returning results, so don't wait for it to be done
-        # (2) don't cancel the task so that it will populate the ES cache for subsequent counts
-        maybe_count = None
+        if len(hits) > limit and limit < max_result_window:
+            if hits and (sort_array := hits[limit - 1].get("sort")):
+                next_token = urlsafe_b64encode(
+                    ",".join([str(x) for x in sort_array]).encode()
+                ).decode()
+
+        matched = es_response["hits"]["total"]["value"]
```

Comment:
In case the count is not exact (over 10,000) and the count task is not ready yet, I would prefer to get no count instead of a wrong one.
Should be able to fix this like this:
```python
matched = es_response["hits"]["total"]["value"] if es_reponse["hits"]["total"]["relation"] == "eq" else None
```

Reply:
It makes sense to me